pkgs/development/libraries/qt-6/default.nix

```diff
@@ -185,9 +185,28 @@ let
       };
 
       wrapQtAppsHook = callPackage
-        ({ makeBinaryWrapper }: makeSetupHook
+        ({ makeBinaryWrapper, runCommand, qtbase, qtwayland }: makeSetupHook
           {
             name = "wrap-qt6-apps-hook";
+            propagatedBuildInputs = [ makeBinaryWrapper ];
+              depsTargetTargetPropagated = [
+              (runCommand "${qtbase.__spliced.targetTarget.name or qtbase.name}-only-plugins" { } ''
+                mkdir -p "$out/lib/qt-6/plugins"
+                ln -s "${qtbase.__spliced.targetTarget or qtbase}/lib/qt-6/plugins" "$out/lib/qt-6/plugins"
```

Comment:
Also, I'm not sure it matters here, but generally we could also want to propagate $out/lib/qt-6/qml.